src/ipc/ipc-common.c

```diff
@@ -182,6 +182,19 @@ void ipc_send_queued_msg(void)
 	k_spin_unlock(&ipc->lock, key);
 }
 
+static void schedule_ipc_worker(void)
+{
+	/*
+	 * note: in XTOS builds, this is handled in
+	 * task_main_primary_core()
+	 */
+#ifdef __ZEPHYR__
+	struct ipc *ipc = ipc_get();
+
+	k_work_schedule(&ipc->z_delayed_work, K_USEC(IPC_PERIOD_USEC));
```

Comment:
so you'd retry every 100us?.. I think previously `IPC_PERIOD_USEC` was actually unused - it was "used" as a task period for the IPC task, but the IPC task was an EDF task, and the EDF scheduler ignores period...

Reply:
@lyakh yes. this is fairly reasonable as we are mostly waiting for the kernel side to ACK the previous message and this should be very fast. OTOH, we could also trigger resend from the IPC driver and use a much slower timer to handle any lingering cases (maybe just to log an error). I can see from history that there was something like this in the old times, but Keyon removed it. Downside is that this needs to be added to all IPC drivers, while this logic is common for all.